src/multitaper.jl

```diff
@@ -0,0 +1,518 @@
+            input[i] = config.window[i, j] * signal[i]
+        end
+        # zero-pad
+        input[(length(signal) + 1):(config.nfft)] .= 0
+
+        mul!(output[:, j], config.plan, input)
+    end
+    return nothing
+end
+
+"""
+    mt_pgram!(output::AbstractVector, signal::AbstractVector, config::MTConfig) -> Periodogram
+
+Computes a multitapered periodogram with parameters specifed by `config`,
+storing the output in `output`.
+
+* `signal::AbstractVector`: should be of length `config.n_samples`
+* `output::AbstractVector`: should be of length `config.n_freq_bins`
+"""
+function mt_pgram!(output::AbstractVector, signal::AbstractVector, config::MTConfig)
+    if length(output) != config.n_freq_bins
+        throw(DimensionMismatch("""Expected `output` to be of length `config.n_freq_bins`;
+                got `length(output) = $(length(output)) and `config.n_freq_bins` = $(config.n_freq_bins)"""))
+    end
+    if length(signal) != config.n_samples
+        throw(DimensionMismatch("""Expected `signal` to be of length `config.n_samples`;
+                got `length(signal) == $(length(signal)) and `config.n_samples` = $(config.n_samples)"""))
+    end
+    tmp = config.fft_output_tmp
+    tapered_spectra!(tmp, signal, config)
+
+    output .= 0
+    for j in 1:(config.ntapers)
+        fft2pow!(output, @view(tmp[:, j]), config.nfft, config.r, config.onesided)
+    end
+    return Periodogram(output, config.freq)
+end
+
+#####
+##### Multitapered spectrogram
+#####
+
+struct MTSpectrogramConfig{T,C<:MTConfig{T}}
+    n_samples::Int
+    n_overlap_samples::Int
+    n_time_points::Int
+    time::FloatRange{Float64}
+    mt_config::C
+end
+
+"""
+    MTSpectrogramConfig{T}(n_samples, n_overlap_samples, mt_config::MTConfig{T}) where {T}
+    MTSpectrogramConfig{T}(n_samples, samples_per_window, n_overlap_samples; fs=1, kwargs...) where {T}
+
+Creates a `MTSpectrogramConfig` which holds configuration and temporary variables for [`mt_spectrogram!`](@ref).
+Any keyword arguments accepted by [`MTConfig`](@ref) may be passed here, or an `MTConfig` object itself.
+"""
+function MTSpectrogramConfig{T}(n_samples, n_overlap_samples, mt_config::MTConfig{T}) where {T}
+    samples_per_window = mt_config.n_samples
+    fs = mt_config.fs
+    # Dimensions of the spectrogram power matrix
+    n_time_points = div(n_samples - samples_per_window,
+                        samples_per_window - n_overlap_samples) + 1
+
+    window_duration = samples_per_window / fs
+    overlap_duration = n_overlap_samples / fs
+
+    hop = window_duration - overlap_duration
+    time = range(window_duration / 2; step=hop, length=n_time_points)
+    return MTSpectrogramConfig{T,typeof(mt_config)}(n_samples, n_overlap_samples,
+                                                              n_time_points, time,
+                                                              mt_config)
+end
+
+function MTSpectrogramConfig{T}(n_samples, samples_per_window, n_overlap_samples;
```

Comment:
order of `samples_per_window, n_overlap_samples` chosen to match `n, n_overlap` ordering of `spectrogram` (and `mt_spectrogram`)